### DISCUSSION TITLE
Adding a `chezmoi-autopush` bash script gives an error: `cannot add chezmoi file to chezmoi`

### DISCUSSION THREAD

**Discussion by gerardnico** [twpayne/chezmoi#3930] [Q&A]
Hallo,

I have written a little bash script named `chezmoi-autopush` to add, commit and push the git repository of chezmoi in the `~/bin` directory.

The content is:
```bash
chezmoi git add .
chezmoi git -- commit -m "$1"
chezmoi git push -u origin main
```

But I got an error when I try to add it:
```bash
chezmoi add "~/bin/chezmoi-autopush"
```
```
chezmoi: /home/nico/bin/chezmoi-autopush: cannot add chezmoi file to chezmoi (/home/nico/bin/chezmoi is protected)
```

The error comes from this [condition](https://github.com/twpayne/chezmoi/blob/0355a62a8afdb820ffa22291e56e0dafec6f49c0/internal/chezmoi/sourcestate.go#L390)

Can I solve this error without rewriting the name of the bash script? 
How can I make this path unprotected? 

Thank you

**Comment by twpayne** [comment]
Why not use chezmoi's [existing autopush support](https://www.chezmoi.io/user-guide/daily-operations/#automatically-commit-and-push-changes-to-your-repo)?

**Reply by gerardnico** [reply]
Man. I'm lost in the notification of GitHub. Sorry for the late answer. I just didn't see it.

I came back to give a solution. Namely installing it in another location
```bash
sudo sh -c "$(curl -fsLS get.chezmoi.io)" -- -b /usr/local/bin
```

Why not use chezmoi's existing support?
Because I may modify multiple files for one change and I want to keep the atomicity not at the file level but at the change level.

Thanks for the answer.
